docs/posts/ibisml/index.qmd

```diff
@@ -0,0 +1,897 @@
+    ],
+    "depth_1": [
+        read_and_process_files(
+            TRAIN_DIR / "train_applprev_1_*.parquet", 1, is_regex=True
+        ),
+        read_and_process_files(TRAIN_DIR / "train_tax_registry_a_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_tax_registry_b_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_tax_registry_c_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_credit_bureau_b_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_other_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_person_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_deposit_1.parquet", 1),
+        read_and_process_files(TRAIN_DIR / "train_debitcard_1.parquet", 1),
+    ],
+    "depth_2": [
+        read_and_process_files(TRAIN_DIR / "train_credit_bureau_b_2.parquet", 2),
+    ],
+}
+# We won't be submitting the predictions, so let's comment out the test data.
+# test_data_store = {
+#     "df_base": read_and_process_files(TEST_DIR / "test_base.parquet"),
+#     "depth_0": [
+#         read_and_process_files(TEST_DIR / "test_static_cb_0.parquet"),
+#         read_and_process_files(TEST_DIR / "test_static_0_*.parquet", is_regex=True),
+#     ],
+#     "depth_1": [
+#         read_and_process_files(TEST_DIR / "test_applprev_1_*.parquet", 1, is_regex=True),
+#         read_and_process_files(TEST_DIR / "test_tax_registry_a_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_tax_registry_b_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_tax_registry_c_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_credit_bureau_b_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_other_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_person_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_deposit_1.parquet", 1),
+#         read_and_process_files(TEST_DIR / "test_debitcard_1.parquet", 1),
+#     ],
+#     "depth_2": [
+#         read_and_process_files(TEST_DIR / "test_credit_bureau_b_2.parquet", 2),
+#     ]
+# }
+```
+
+Join all features data to base table:
+
+```{python}
+#| code-fold: true
+#| code-summary: "Define function to join feature tables to base table"
+def join_data(df_base, depth_0, depth_1, depth_2):
+    for i, df in enumerate(depth_0 + depth_1 + depth_2):
+        df_base = df_base.join(
+            df, "case_id", how="left", rname="{name}_right" + f"_{i}"
+        )
+    return df_base
+```
+
+Generate train and test datasets:
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to generate train and test datasets"
+df_train = join_data(**train_data_store)
+# df_test = join_data(**test_data_store)
+print(f"There is {df_train.count()} rows and {len(df_train.columns)} columns")
+```
+### Select features
+Since there are ~370 features, we'll only select part of good ones for this post:
```

Comment:
```suggestion
Since there are ~370 features, we'll only select a few of the good ones for this post:
```
Also, what does "good" mean? Maybe "informative" or "useful" would be a better word?